File: app/src/main/java/com/example/my_first_calculator/MainActivity.java
package com.example.my_first_calculator;

import androidx.appcompat.app.AppCompatActivity;
import android.view.inputmethod.InputMethodManager;
import android.content.Context;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button addButton = findViewById(R.id.addBtn);
        Button substractButton = findViewById(R.id.substractBtn);
        Button divideButton = findViewById(R.id.divideBtn);
        Button multiplyButton = findViewById(R.id.multiplyBtn);

        EditText numero1 = findViewById(R.id.numero1);
        EditText numero2 = findViewById(R.id.numero2);
        TextView resultText = findViewById(R.id.result);

        InputMethodManager inputManager = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);




        addButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(numero1.getText().toString().isEmpty() || numero2.getText().toString().isEmpty()){
                    Toast.makeText(getBaseContext(), "falta un numero", Toast.LENGTH_LONG).show();
                    return;
                }
                int num1 = Integer.parseInt(numero1.getText().toString());
                int num2 = Integer.parseInt(numero2.getText().toString());


                int result = calculate(num1, num2, 1);

                if(inputManager != null) {
                    View currentFocusedView = getCurrentFocus();
                    if (currentFocusedView != null) {
                        inputManager.hideSoftInputFromWindow(currentFocusedView.getWindowToken(), InputMethodManager.HIDE_NOT_ALWAYS);
                    }
                }

                resultText.setText("El resultado es: " + result);
            }
        });

        substractButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(numero1.getText().toString().isEmpty() || numero2.getText().toString().isEmpty()){
                    Toast.makeText(getBaseContext(), "falta un numero", Toast.LENGTH_LONG).show();
                    return;
                }

                int num1 = Integer.parseInt(numero1.getText().toString());
                int num2 = Integer.parseInt(numero2.getText().toString());


                int result = calculate(num1, num2, 2);

                if(inputManager != null) {
                    View currentFocusedView = getCurrentFocus();
                    if (currentFocusedView != null) {
                        inputManager.hideSoftInputFromWindow(currentFocusedView.getWindowToken(), InputMethodManager.HIDE_NOT_ALWAYS);
                    }
                }

                resultText.setText("El resultado es: " + result);
            }
        });

        multiplyButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(numero1.getText().toString().isEmpty() || numero2.getText().toString().isEmpty()){
                    Toast.makeText(getBaseContext(), "falta un numero", Toast.LENGTH_LONG).show();
                    return;
                }

                int num1 = Integer.parseInt(numero1.getText().toString());
                int num2 = Integer.parseInt(numero2.getText().toString());


                int result = calculate(num1, num2, 4);

                if(inputManager != null) {
                    View currentFocusedView = getCurrentFocus();
                    if (currentFocusedView != null) {
                        inputManager.hideSoftInputFromWindow(currentFocusedView.getWindowToken(), InputMethodManager.HIDE_NOT_ALWAYS);
                    }
                }

                resultText.setText("El resultado es: " + result);
            }
        });

        divideButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(numero1.getText().toString().isEmpty() || numero2.getText().toString().isEmpty()){
                    Toast.makeText(getBaseContext(), "falta un numero", Toast.LENGTH_LONG).show();
                    return;
                }

                int num1 = Integer.parseInt(numero1.getText().toString());
                int num2 = Integer.parseInt(numero2.getText().toString());


                int result = calculate(num1, num2, 3);

                if(inputManager != null) {
                    View currentFocusedView = getCurrentFocus();
                    if (currentFocusedView != null) {
                        inputManager.hideSoftInputFromWindow(currentFocusedView.getWindowToken(), InputMethodManager.HIDE_NOT_ALWAYS);
                    }
                }

                resultText.setText("El resultado es: " + result);
            }
        });

    }

    private int calculate(int num1, int num2, int operand){
        int result = 0;
            switch (operand) {
                //sum
                case 1:
                    result = num1 + num2;
                    break;
                //subtraction
                case 2:
                    result = num1 - num2;
                    break;
                case 3:
                    if (num2 == 0) {
                        Toast.makeText(getBaseContext(), "no se puede dividir por 0", Toast.LENGTH_LONG).show();
                        break;
                    }
                    result = num1 / num2;
                    break;
                case 4:
                    result = num1 * num2;


            }
            
            return result;
    }
}

File: app/src/main/java/com/example/my_first_calculator/MainActivity.kt
package com.example.my_first_calculator

import android.os.Bundle
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val addButton = findViewById<Button>(R.id.addBtn)
        val substractButton = findViewById<Button>(R.id.substractBtn)
        val divideButton = findViewById<Button>(R.id.divideBtn)
        val multiplyButton = findViewById<Button>(R.id.multiplyBtn)
        val numero1 = findViewById<EditText>(R.id.numero1)
        val numero2 = findViewById<EditText>(R.id.numero2)
        val resultText = findViewById<TextView>(R.id.result)
        val inputManager = getSystemService(INPUT_METHOD_SERVICE) as InputMethodManager
        addButton.setOnClickListener(View.OnClickListener {
            if (numero1.text.toString().isEmpty() || numero2.text.toString().isEmpty()) {
                Toast.makeText(baseContext, "falta un numero", Toast.LENGTH_LONG).show()
                return@OnClickListener
            }
            val num1 = numero1.text.toString().toInt()
            val num2 = numero2.text.toString().toInt()
            val result = calculate(num1, num2, 1)
            resultText.text = "El resultado es: $result"
        })
        substractButton.setOnClickListener(View.OnClickListener {
            if (numero1.text.toString().isEmpty() || numero2.text.toString().isEmpty()) {
                Toast.makeText(baseContext, "falta un numero", Toast.LENGTH_LONG).show()
                return@OnClickListener
            }
            val num1 = numero1.text.toString().toInt()
            val num2 = numero2.text.toString().toInt()
            val result = calculate(num1, num2, 2)
            resultText.text = "El resultado es: $result"
        })
        multiplyButton.setOnClickListener(View.OnClickListener {
            if (numero1.text.toString().isEmpty() || numero2.text.toString().isEmpty()) {
                Toast.makeText(baseContext, "falta un numero", Toast.LENGTH_LONG).show()
                return@OnClickListener
            }
            val num1 = numero1.text.toString().toInt()
            val num2 = numero2.text.toString().toInt()
            val result = calculate(num1, num2, 4)
             resultText.text = "El resultado es: $result"
        })
        divideButton.setOnClickListener(View.OnClickListener {
            if (numero1.text.toString().isEmpty() || numero2.text.toString().isEmpty()) {
                Toast.makeText(baseContext, "falta un numero", Toast.LENGTH_LONG).show()
                return@OnClickListener
            }
            val num1 = numero1.text.toString().toInt()
            val num2 = numero2.text.toString().toInt()
            val result = calculate(num1, num2, 3)
             resultText.text = "El resultado es: $result"
        })
    }

    private fun calculate(num1: Int, num2: Int, operand: Int): Int {
        var result = 0
        when (operand) {
            1 -> result = num1 + num2
            2 -> result = num1 - num2
            3 -> {
                if (num2 == 0) {
                    Toast.makeText(baseContext, "no se puede dividir por 0", Toast.LENGTH_LONG).show()
                    throw IllegalArgumentException("no se puede dividir por 0")
                }
                result = num1 / num2
            }

            4 -> result = num1 * num2
        }
        return result
    }
}